tasks/post-to-cmr/tests/cmr-test.js

```diff
@@ -128,9 +129,61 @@ test.serial('postToCMR skips CMR step if the metadata file uri is missing', asyn
     }]
   }];
 
-  const output = await postToCMR(newPayload);
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+  }
+});
+
+test.serial('postToCMR throws an error if any granule is missing a Meta file', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = {
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.xml`
+    }]
+  };
+  newPayload.input.granules.push(newGranule);
+
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+    t.is(error.message, (`CMR Meta file not found for granule ${newGranule.granuleId}`));
+  }
+});
+
+test.serial('postToCMR continues without Meta file if there is metaCheck flag', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = [{
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.xml`
+    }]
+  }];
+  newPayload.input.granules = newGranule;
+  newPayload.config.metaCheck = true;
+  const granuleId = newPayload.input.granules[0].granuleId;
+  const key = `${granuleId}.cmr.xml`;
 
-  t.is(output.granules[0].cmrLink, undefined);
+  sinon.stub(cmrjs.CMR.prototype, 'ingestGranule').callsFake(() => ({
```

Comment:
Is this stubbing and staging of a CMR XML file necessary for `postToCmr` to complete? I was initially confused here because you're staging a CMR XML file, so it seems like this test is including a metadata file, but then I noticed that file isn't referenced in the payload

Reply:
Oh you are correct, not necessary.

Reply:
OK. If it's not needed can we remove it?

Reply:
Oh we need it in this test. starting at line 186, because there is 1 granule being processed with a metadata file. The test on line 159 doesn't need it because there are 0 metadata files.